docs/multi-agent-applications.md

```diff
@@ -0,0 +1,333 @@
+from pydantic_ai_examples.sql_gen import system_prompt
+
+# Multi-agent Applications
+
+There are roughly four levels of complexity when building applications with PydanticAI:
+
+1. Single agent workflows — what most of this documentation covers
+2. [Agent delegation](#agent-delegation) — agents using another agent via tools
+3. [Programmatic agent hand-off](#programmatic-agent-hand-off) — one agent runs, then application code calls another agent
+4. [Graph based control flow](#pydanticai-graphs) — for the most complex cases, graph and a state machine can be used to control the execution of multiple agents
+
+Of course, you can combine multiple strategies in a single application.
+
+## Agent Delegation
+
+The agent delegates work to another agent, but then takes back control when that agent finishes.
+
+Since agents are stateless and designed to be global, you do not need to include the agent itself in agent [dependencies](dependencies.md).
+
+When doing so, you'll generally want to pass [`ctx.usage`][pydantic_ai.RunContext.usage] to the [`usage`][pydantic_ai.Agent.run] keyword argument of delegate agent (the agent called from within a tool) run so usage within that run counts towards the total usage of a parent agent run.
+
+!!! Multiple models
+    Agent delegation doesn't need to use the same model for each agent. If you choose to use different models within a run, calculating the monetary cost from the final [`result.usage()`][pydantic_ai.result.RunResult.usage] of the run will not be possible, but you can still use [`UsageLimits`][pydantic_ai.usage.UsageLimits] to avoid unexpected costs.
+
+```python {title="agent_delegation_simple.py"}
+from pydantic_ai import Agent, RunContext
+from pydantic_ai.usage import UsageLimits
+
+joke_agent = Agent(
+    'openai:gpt-4o',
+    system_prompt=(
+        'Use the "joke_factory" to generate some jokes, then choose the best. '
+        'You must return just a single joke.'
+    ),
+)
+
+
+@joke_agent.tool
+async def joke_factory(ctx: RunContext[None], count: int) -> list[str]:
+    r = await delegate_agent.run(f'Please generate {count} jokes.', usage=ctx.usage)
+    return r.data
+
+
+delegate_agent = Agent('gemini-1.5-flash', result_type=list[str])
+
+result = joke_agent.run_sync(
+    'Tell me a joke.', usage_limits=UsageLimits(request_limit=5, total_tokens_limit=300)
+)
+print(result.data)
+#> Did you hear about the toothpaste scandal? They called it Colgate.
+print(result.usage())
+"""
+Usage(
+    requests=3, request_tokens=204, response_tokens=24, total_tokens=228, details=None
+)
+"""
+```
+
+_(This example is complete, it can be run "as is")_
```

Comment:
Seeing a lot of these across the docs - why do we include this? Shouldn't this be the assumption?

Reply:
only in pydantic, it's not the case in most libraries (naming no names 👀 ), happy to automate it somehow, but I think it's work including in the final docs.